src/transformers/models/ibert/utils/quant_modules.py

```diff
@@ -0,0 +1,538 @@
+            quant_act_int = self.act_function(x, self.activation_bit, \
+                    self.percentile, self.act_scaling_factor)
+        else:
+            quant_act_int = fixedpoint_mul.apply(
+                    x, pre_act_scaling_factor, 
+                    self.activation_bit, self.quant_mode, 
+                    self.act_scaling_factor, 
+                    identity, identity_scaling_factor)
+
+        correct_output_scale = self.act_scaling_factor.view(-1)
+
+        return quant_act_int * correct_output_scale, self.act_scaling_factor
+
+
+class QuantLinear(Module):
+    """
+    Class to quantize weights of given Linear layer
+    
+    Parameters:
+    ----------
+    weight_bit : int
+        Bitwidth for quantized weights.
+    bias_bit : int, default None
+        Bitwidth for quantized bias.
+    per_channel : bool, default False
+        Whether to use channel-wise quantization.
+    quant_mode : bool, default False
+        The mode for quantization. True for quantization.
+    """
+    def __init__(self,
+                 weight_bit,
+                 bias_bit=None,
+                 per_channel=False,
+                 quant_mode=False):
+        super(QuantLinear, self).__init__()
+        self.weight_bit = weight_bit
+        self.quant_mode = quant_mode
+        self.per_channel = per_channel
+        self.bias_bit = bias_bit
+        self.quantize_bias = (False if bias_bit is None else True)
+        self.quant_mode = quant_mode
+        self.percentile_mode = False
+
+        if not self.quant_mode:
+            self.weight_function = None
+        else:
+            self.weight_function = SymmetricQuantFunction.apply
+
+    def __repr__(self):
+        s = super(QuantLinear, self).__repr__()
+        s = "(" + s + " weight_bit={}, quant_mode={})".format(
+            self.weight_bit, self.quant_mode)
+        return s
+
+    def set_param(self, linear):
+        self.in_features = linear.in_features
+        self.out_features = linear.out_features
+        self.weight = Parameter(linear.weight.data.clone())
+        self.register_buffer('fc_scaling_factor', torch.zeros(self.out_features))
+        self.register_buffer('weight_integer', torch.zeros_like(self.weight))
+        try:
+            self.bias = Parameter(linear.bias.data.clone())
+            self.register_buffer('bias_integer', torch.zeros_like(self.bias))
+        except AttributeError:
+            self.bias = None
+            self.bias_integer = None
+
+    def forward(self, x, prev_act_scaling_factor=None):
+        """
+        using quantized weights to forward activation x
+        """
+        if not self.quant_mode:
+            return F.linear(x, weight=self.weight, bias=self.bias), None
+
+        # assert that prev_act_scaling_factor is a scalar tensor
+        # i.e., it is not channel-wise quantized
+        assert prev_act_scaling_factor is not None and \
+              prev_act_scaling_factor.shape == (1,) 
```

Comment:
Please put an error message. Empty asserts make it harder to debug when there is a problem.